### PR TITLE
Added more possible streetalike keys and some new keys

### LINKED ISSUE #23: Missing Adressdetails Entries
It occurs that when querying from nominatim or opencage some addressdetails are missing compared to the graphhopper geocoding api.

A sample request for: `münchen%20prinzregentenstraße%20122` returns:
GraphHopper:
```
{
osm_id: 1217311546,
osm_type: "N",
country: "Deutschland",
osm_key: "amenity",
housenumber: "122",
city: "München",
street: "Prinzregentenstraße",
osm_value: "restaurant",
postcode: "81677",
name: "Sushi Oji",
state: "Bayern",
point: {
lng: 11.6154759,
lat: 48.1379734
}
}
```
Nominatim:
```
{
point: {
lat: 48.1379734,
lng: 11.6154759
},
name: "Sushi Oji, Prinzregentenstraße 122, 81677 München, Deutschland",
country: "Deutschland",
city: "München",
state: "Bayern",
osm_id: 1217311546,
osm_type: "N",
house_number: "122"
}
```
Opencagedata:
```
{
point: {
lat: 48.1379734,
lng: 11.6154759
},
name: "Sushi Oji, Prinzregentenstraße 122, 81677 München, Deutschland",
country: "Deutschland",
city: "München",
state: "Bayern",
osm_id: 1217311546,
osm_type: "N",
house_number: "122"
}
```

Missing right now is:
- street
- postcode
- osm_value
- osm_key

### PR DESCRIPTION
**Pull request by boldtrn**
This PR intents to fix #23. 

I added the following keys **postcode** and **osm_value**. Also I added more possible street name keys. It seems that nominatim and opencagedata return a diverse set of streetname keys, not sure if there is a complete list of keys out there, it seems to be different from the osm highways tag.

I have not aded the osm_key as opencagedata seems to not return the osm_key, but I can add it for Nominatim?

@karussell Would you mind to have a look at my changes?

### CODE REVIEW

File: src/main/java/com/graphhopper/converter/api/NominatimEntry.java [hunk 1]
```diff
@@ -18,6 +18,9 @@
     private String displayName;
     private String classString;
 
+    // This is not the omsType
+    private String type;
```
Comment (karussell) [inline src/main/java/com/graphhopper/converter/api/NominatimEntry.java:22]: Would you mind to add an example in the comments here? Is this node vs. way?

File: src/main/java/com/graphhopper/converter/api/NominatimEntry.java [hunk 2]
```diff
@@ -43,7 +46,13 @@ public String getStreetName() {
...
             return this.address.pedestrian;
         }
-        return null;
+        if (this.address.path != null) {
+            return this.address.path;
+        }
+        if (this.address.footway != null) {
+            return this.address.footway;
+        }
+        return this.address.construction;
```
Comment (karussell) [inline src/main/java/com/graphhopper/converter/api/NominatimEntry.java]: Would you mind to move this into a new method address.getStreetName?

File: src/main/java/com/graphhopper/converter/api/NominatimEntry.java [hunk 3]
```diff
@@ -141,12 +160,22 @@ public Address() {
...
+        @JsonProperty
+        public String road;
+        @JsonProperty
+        public String pedestrian;
+        @JsonProperty
+        public String path;
+        @JsonProperty
+        public String footway;
+        @JsonProperty
+        public String construction;
```
Comment (karussell) [inline src/main/java/com/graphhopper/converter/api/NominatimEntry.java]: Oh, this is ugly ... and I also did not find documentation for this. One further tag you can add is cycleway: http://nominatim.openstreetmap.org/search.php?q=bayreuth%20fahrradhighway&polygon=1&format=json&addressdetails=1

File: src/main/java/com/graphhopper/converter/api/OpenCageDataEntry.java [hunk 4]
```diff
@@ -77,6 +73,20 @@ public OCDComponents() {
...
+        @JsonProperty
+        public String road;
+        @JsonProperty
+        public String pedestrian;
+        @JsonProperty
+        public String path;
+        @JsonProperty
+        public String footway;
+        @JsonProperty
+        public String construction;
```
Comment (karussell) [inline src/main/java/com/graphhopper/converter/api/OpenCageDataEntry.java]: Can we somehow reuse the Address stuff here?

**Review by karussell — CHANGES_REQUESTED** [review]
Thanks! Just some minor comments ...

**Comment by karussell** [comment]
How do we handle the different name entry? E.g. we could parse "restaurant":"Sushi Oji" and use this as name but we would overwrite the full display name: "Sushi Oji, Prinzregentenstraße 122, 81677 München, Deutschland" and potentially break existing apps... hmmh

File: src/main/java/com/graphhopper/converter/api/NominatimEntry.java (hunk 2 above)
Comment (boldtrn) [reply src/main/java/com/graphhopper/converter/api/NominatimEntry.java]: Not at all, I had the same thought :)

File: src/main/java/com/graphhopper/converter/api/NominatimEntry.java (hunk 3 above)
Comment (boldtrn) [reply src/main/java/com/graphhopper/converter/api/NominatimEntry.java]: Thanks, added cycleway.

File: src/main/java/com/graphhopper/converter/api/OpenCageDataEntry.java (hunk 4 above)
Comment (boldtrn) [reply src/main/java/com/graphhopper/converter/api/OpenCageDataEntry.java]: I wans't sure if this works with Jackson, but I was able to move all duplicated entries to an abstract class.

**Comment by boldtrn** [comment]
> How do we handle the different name entry?

I always thought this would be one of the benefits of using a different geocoder. For example having the more detailed display_name of Nominatim.

**Comment by boldtrn** [comment]
I updated my branch and fixed your requested changes.

**Review by karussell — APPROVED** [review]

**Comment by karussell** [comment]
> > How do we handle the different name entry?

> I always thought this would be one of the benefits of using a different geocoder. For example having the more detailed display_name of Nominatim.

Using different geocoders will give you different quality but the fields should be the same information. So if it is a POI the `name` field should contain just its name not the full address and if it is a street, then it could just contain the street name etc. having a field `all_name` or `display_name` with all information could be a nice addition though, which we would need for photon too then.

**Pull request merged**